Tools/autotest/autotest.py

```diff
@@ -153,94 +156,30 @@ def alarm_handler(signum, frame):
-parser.add_option("--skip", type='string', default='', help='list of steps to skip (comma separated)')
-parser.add_option("--list", action='store_true', default=False, help='list the available steps')
-parser.add_option("--viewerip", default=None, help='IP address to send MAVLink and fg packets to')
-parser.add_option("--map", action='store_true', default=False, help='show map')
-parser.add_option("--experimental", default=False, action='store_true', help='enable experimental tests')
-parser.add_option("--timeout", default=3000, type='int', help='maximum runtime in seconds')
-parser.add_option("--valgrind", default=False, action='store_true', help='run ArduPilot binaries under valgrind')
-parser.add_option("--gdb", default=False, action='store_true', help='run ArduPilot binaries under gdb')
-parser.add_option("--debug", default=False, action='store_true', help='make built binaries debug binaries')
-parser.add_option("-j", default=None, type='int', help='build CPUs')
-parser.add_option("--frame", type='string', default=None, help='specify frame type')
-
-opts, args = parser.parse_args()
-
-
-steps = [
-    'prerequisites',
-    'build.All',
-    'build.Binaries',
-    # 'build.DevRelease',
-    'build.Examples',
-    'build.Parameters',
-
-    'build.ArduPlane',
-    'defaults.ArduPlane',
-    'fly.ArduPlane',
-    'fly.QuadPlane',
-
-    'build.APMrover2',
-    'defaults.APMrover2',
-    'drive.APMrover2',
-
-    'build.ArduCopter',
-    'defaults.ArduCopter',
-    'fly.ArduCopter',
-
-    'build.Helicopter',
-    'fly.CopterAVC',
-
-    'build.AntennaTracker',
-    
-    'build.ArduSub',
-    'defaults.ArduSub',
-    'dive.ArduSub',
-
-    'convertgpx',
-    ]
-
-skipsteps = opts.skip.split(',')
-
-# ensure we catch timeouts
-signal.signal(signal.SIGALRM, alarm_handler)
-signal.alarm(opts.timeout)
-
-if opts.list:
-    for step in steps:
-        print(step)
-    sys.exit(0)
 
+    sys.exit(1)
 
 def skip_step(step):
     """See if a step should be skipped."""
     for skip in skipsteps:
         if fnmatch.fnmatch(step.lower(), skip.lower()):
             return True
+
     return False
 
+__bin_names = {
+    "ArduCopter" : "arducopter",
+    "ArduPlane" : "arduplane",
+    "APMrover2" : "ardurover",
+    "AntennaTracker" : "antennatracker",
+    "CopterAVC" : "arducopter-heli",
+    "QuadPlane" : "arduplane",
+    "ArduSub" : "ardusub"
+}
 
 def binary_path(step, debug=False):
-    if step.find("ArduCopter") != -1:
-        binary_name = "arducopter"
-    elif step.find("ArduPlane") != -1:
-        binary_name = "arduplane"
-    elif step.find("APMrover2") != -1:
-        binary_name = "ardurover"
-    elif step.find("AntennaTracker") != -1:
-        binary_name = "antennatracker"
-    elif step.find("CopterAVC") != -1:
-        binary_name = "arducopter-heli"
-    elif step.find("QuadPlane") != -1:
-        binary_name = "arduplane"
-    elif step.find("ArduSub") != -1:
-        binary_name = "ardusub"
+    if step in __bin_names:
```

Comment:
`__bin_names` doesn't contain step names... the old code was doing substring matches (against things like `fly.ArduCopter`)